### PR TITLE
chore(all): fix doc typos and apply basic lint recommendations

### PR DESCRIPTION
**Pull request by michaelbrewer**
## Description of your changes

- Simplify comparisons
- Fix gramma or spelling on doc strings
- Fix doc string typos

### How to verify this change

Run pre commit and post commit hooks

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [X] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [X] I have performed a *self-review* of my own code
- [X] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [X] I have made corresponding changes to the *documentation*
- [X] My changes generate *no new warnings*
- [X] The *code coverage* hasn't decreased
- [X] I have *added tests* that prove my change is effective and works
- [X] New and existing *unit tests pass* locally and in Github Actions
- [X] Any *dependent changes have been merged and published* in downstream module
- [X] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/tracing/src/Tracer.ts [hunk 1]
```diff
@@ -140,12 +140,12 @@ class Tracer implements TracerInterface {
     * @param error - Error to serialize as metadata
     */
   public addErrorAsMetadata(error: Error): void {
-    if (this.tracingEnabled === false) {
+    if (!this.tracingEnabled) {
```
Comment (dreamorosi) [inline packages/tracing/src/Tracer.ts:143]: Nice, we have another PR (#442) that is moving all these `this.tracingEnabled` to use `this.isTracingEnabled()`. Depending on which one gets merged first we might have to solve some conflicts but no big deal.

**Review by dreamorosi — APPROVED** [review]

File: packages/tracing/src/Tracer.ts (hunk 1 above)
Comment (michaelbrewer) [reply packages/tracing/src/Tracer.ts:143]: Sure.

Comment (michaelbrewer) [reply packages/tracing/src/Tracer.ts]: @dreamorosi ok i have merged in the charges

**Review by dreamorosi — APPROVED** [review]
Thanks for the PR!

File: packages/tracing/src/Tracer.ts [hunk 2]
```diff
@@ -255,7 +255,7 @@ class Tracer implements TracerInterface {
    * @returns service - Instrumented AWS SDK v2 client
    */
   public captureAWSClient<T>(service: T): T {
-    if (this.isTracingEnabled() === false) return service;
+    if (!this.isTracingEnabled()) return aws;
```
Comment (dreamorosi) [inline packages/tracing/src/Tracer.ts:258]: This return should be `service`, not `aws`:

```suggestion
    if (!this.isTracingEnabled()) return service;
```

File: packages/tracing/src/Tracer.ts [hunk 3]
```diff
@@ -285,7 +285,7 @@ class Tracer implements TracerInterface {
    * @returns service - Instrumented AWS SDK v3 client
    */
   public captureAWSv3Client<T>(service: T): T {
-    if (this.isTracingEnabled() === false) return service;
+    if (!this.isTracingEnabled()) return aws;
```
Comment (dreamorosi) [inline packages/tracing/src/Tracer.ts:288]:
```suggestion
    if (!this.isTracingEnabled()) return service;
```

**Review by dreamorosi — CHANGES_REQUESTED** [review]
Two minor changes

**Comment by dreamorosi** [comment]
@michaelbrewer sorry for the double review, I was looking at it again and noticed two lines that I think need attention.

**Comment by michaelbrewer** [comment]
> @michaelbrewer sorry for the double review, I was looking at it again and noticed two lines that I think need attention.

sorry i probably messed up with doing the merge conflicts. It should be fine now.

**Review by flochaz — APPROVED** [review]

**Pull request merged**